apps/etterlatte-hendelser-joark/src/main/kotlin/joarkhendelser/JoarkHendelseHandler.kt

```diff
@@ -1,31 +1,142 @@
+                //  Må vi lage støtte for ORGNR...?
+                logger.error("Journalpost med id=$journalpostId har brukerId av typen ${BrukerIdType.ORGNR}")
+                return
+            }
+
+            val pdlIdentifikator =
+                pdlKlient.hentPdlIdentifikator(journalpost.bruker.id)
+
+            val ident =
+                when (pdlIdentifikator) {
+                    is PdlIdentifikator.FolkeregisterIdent -> pdlIdentifikator.folkeregisterident.value
+                    is PdlIdentifikator.Npid -> {
+                        logger.error("Ignorerer journalføringshendelse med NPID=${pdlIdentifikator.npid.ident}")
+                        return
+                    }
+
+                    null -> throw IllegalArgumentException(
+                        "Ident tilknyttet journalpost=$journalpostId er null i PDL – avbryter behandling",
+                    )
+                }
+
+            when (val type = hendelse.hendelsesType) {
+                HendelseType.JOURNALPOST_MOTTATT ->
+                    behandleJournalpost(ident, sakType, hendelse) {
+                        hendelse.lagMerknadFraStatus()
+                    }
+
+                HendelseType.TEMA_ENDRET ->
+                    behandleJournalpost(ident, sakType, hendelse) {
+                        "Tema endret fra ${hendelse.temaGammelt} til ${hendelse.temaNytt}"
+                    }
+
+                // TODO: Må avklare om dette er noe vi faktisk trenger å behandle
+                HendelseType.ENDELIG_JOURNALFOERT ->
+                    behandleJournalpost(ident, sakType, hendelse) {
+                        "Endelig journalføring må vurderes"
+                    }
+
+                // TODO: Må avklare om dette er noe vi faktisk trenger å behandle
+                HendelseType.JOURNALPOST_UTGAATT ->
+                    behandleJournalpost(ident, sakType, hendelse) {
+                        "Journalpost har utgått og må vurderes."
+                    }
+                else -> throw IllegalArgumentException("Journalpost=$journalpostId har ukjent hendelsesType=$type")
+            }
         } catch (e: Exception) {
+            logger.error("Ukjent feil ved behandling av hendelse=${hendelse.hendelsesId}. Se sikkerlogg for mer detaljer.")
+            sikkerlogger().error("Ukjent feil oppsto ved behandling av journalpost for bruker=${journalpost.bruker}: ", e)
             throw e
         }
     }
+
+    private suspend fun behandleJournalpost(
+        ident: String,
+        sakType: SakType,
+        hendelse: JournalfoeringHendelseRecord,
+        oppgaveMerknad: () -> String,
+    ) {
+        val gradering = pdlKlient.hentAdressebeskyttelse(ident)
+        logger.info("Bruker=${ident.maskerFnr()} har gradering $gradering")
+
+        logger.info("Oppretter ny ${sakType.name.lowercase()} for bruker=${ident.maskerFnr()} med gradering=$gradering")
```

Comment:
Den vil vel ikke alltid opprette ny sak? Kan vel være en journalhendelse for en eksisterende?

Reply:
Nei, gammel moro som henger igjen. Skal oppdatere meldingen 👍